bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -1156,6 +1158,10 @@ private LedgerDirsManager.LedgerDirsListener getLedgerDirsListener() {
 
             @Override
             public void diskAlmostFull(File disk) {
+                if (!Objects.equals(disk.getPath(), currentFile.getPath())) {
```

Comment:
let's reduce code duplication and make this block a method

if (!isCurrentFile(disk)) {
 return;
}